### PR TITLE
Cherry pick upstream changes

### PR DESCRIPTION
**Pull request by braem**
Tackles part of https://github.com/ChaosInitiative/Chaos-FGD/issues/19

Cherry picks the bulk of the changes from teamspen's repo up until now.

**- Stuff I didn't pull in -**

https://github.com/TeamSpen210/HammerAddons/commit/cd7264dd438d63e1a6a78c9a4289077d1dece0e8 - dont have this in engine

https://github.com/TeamSpen210/HammerAddons/commit/084c7829aee3d21e60e1a79ad14b12d56ccef31a - added our own `Brightness`/`SetBrightness` field/input as using `renderamt` requires spotlight to be recreated (cant change dynamically).

https://github.com/TeamSpen210/HammerAddons/commit/c10e00af3632466219df08897ec4423df1f05fb4 , https://github.com/TeamSpen210/HammerAddons/commit/6543cef4ebad32f99d56bbadefbcc9f3212fe8f9 - extension KV stuff. Deserves it's own PR if we want it.

https://github.com/TeamSpen210/HammerAddons/commit/04897842673ef4f3bf6d41d647ffc047985a7a4e , https://github.com/TeamSpen210/HammerAddons/commit/7f2850789301fc26a7232bf335331d70588c0661 - clarifying visgroups. will need it's own PR (might not be able to cherrypick with all our changes)

https://github.com/TeamSpen210/HammerAddons/commit/bd308fcb7303fd13657bd8d24b5a5174fef15f50 - have this already (thx goc)

https://github.com/TeamSpen210/HammerAddons/commit/76bfcb584163f5535531098adb71b7b24f34c90a - bye bye min/max dxlevel

https://github.com/TeamSpen210/HammerAddons/commit/5759c7004acc43988a76b2f25f835a3886fdc9b9 - this mistake caused the entity to be missed in our big removal PRs. It's not in our engine so I removed it.

https://github.com/TeamSpen210/HammerAddons/commit/a795341192997ed209d884de4460d31112051295 - wondering what this is for

https://github.com/TeamSpen210/HammerAddons/commit/df576061087637af48eba87952701302adb4e623 , https://github.com/TeamSpen210/HammerAddons/commit/84b6401f9b1d6792079e21476c245bbc6c37810c , https://github.com/TeamSpen210/HammerAddons/commit/14e6df4bb5fb3748426388f29f8d64b9d5530903 - material exclusions, could be useful? Deserves it's own PR

https://github.com/TeamSpen210/HammerAddons/commit/40649e10ef11209247fabc424bb5cdc215fd7a05 , https://github.com/TeamSpen210/HammerAddons/commit/2296e1f7306c87e9047e0c36e3a8730be2b54f52 , https://github.com/TeamSpen210/HammerAddons/commit/69b7137e77810918231076146512e07c2808b3bb , https://github.com/TeamSpen210/HammerAddons/commit/a52c6a1e9246b9b194e7ee2acfefbd8153482849 , https://github.com/TeamSpen210/HammerAddons/commit/bc572cc1e0995d7adfb7912ff62388e3710c2338 - addition of base entity. Will need to be it's own PR as could be more involved.

https://github.com/TeamSpen210/HammerAddons/commit/041232ddcc6594706036784983d2075d5c8a1623 , https://github.com/TeamSpen210/HammerAddons/commit/48b7055bf0f033c1261fc0d00d6b0b45d04be79d , https://github.com/TeamSpen210/HammerAddons/commit/cdc0f24119dfbe9b0badd1de55a67826d0e10426 , https://github.com/TeamSpen210/HammerAddons/commit/6d43f18a1a936c7dc4cfafb6e6f81f53f9a8ce46 , https://github.com/TeamSpen210/HammerAddons/commit/49d26ccee6274696f7256bf81a636c5f0c4a16c1 , https://github.com/TeamSpen210/HammerAddons/commit/e233843457bd8f4b1bd3b3db68218615ddc96091 , https://github.com/TeamSpen210/HammerAddons/commit/dc8b1ca21818ef81780ca93d19fa0bf465db87dc , https://github.com/TeamSpen210/HammerAddons/commit/d6ab8c674d5eda03b4021134c851aad82f2b8a95 , https://github.com/TeamSpen210/HammerAddons/commit/13cbe6320da1552474b2c8d2e9ba2336c9a28d8b , https://github.com/TeamSpen210/HammerAddons/commit/8283234e2449ac10ac163c3390e8e3df7887c9c4 , https://github.com/TeamSpen210/HammerAddons/commit/7bbc3d9635ee2cfe556b3b206b24c1d787a680de - all L4D stuff; dont need

https://github.com/TeamSpen210/HammerAddons/commit/0eaaefed370aeb3399f176d903dd02cf19252acc , https://github.com/TeamSpen210/HammerAddons/commit/aa2e5089a88c4b814e0739404744480217a7adde , https://github.com/TeamSpen210/HammerAddons/commit/e0196d4dca887326a1d8d67023046712b9d3782d , https://github.com/TeamSpen210/HammerAddons/commit/d0b07b66a84758aadbec6dfc3fa112249e1e9f70 - readme changes. Should be in it's own PR as we will be changing the readme anyhow to be better inline with what Chaos is using this repo for (different goal than teamspen).

### CODE REVIEW

**Comment by TeamSpen210** [comment]
The modified Crowbar is used for propcombine, allowing it to automatically decompile the required props directly. The KV extensions I did is to prepare for implementing instance collapsing in srctools - it uses this database also to know how to transform the keys.

File: fgd/point/comp/comp_prop_cable.fgd [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+	u_min(float): "Width Start" : 0.0 : "The distance along the texture to start. 0 is the left/bottom side, 1 is the right/top side. This allows using only part of the texture, if it contains multiple different cable styles."
+	u_max(float): "Width End"   : 1.0 : "The distance along the texture to end. 0 is the left/bottom side, 1 is the right/top side. This allows using only part of the texture, if it contains multiple different cable styles."
+	
+	// A selection of prop_static's keyvalues, excluding some that are rather irrelevant.
+	linedivider_staticprop[!engine](string) readonly : "----------------------------------------------------------------------------------------------------------" : ""
+
+	renderamt[since_L4D](integer) : "Alpha" : 255 : "Alpha of the fade, where 0 = fully transparent and 255 = fully opaque."
+	rendercolor[since_L4D](color255) : "Color (R G B)" : "255 255 255"
+
+	screenspacefade[until_ASW](boolean) : "Screen Space Fade" : 0 : "The method by which the fading distance should be determined. If 'No', the fade distances is the distance from the player's view to the object, in inches. " +
```
Comment (Gocnak) [inline fgd/point/comp/comp_prop_cable.fgd:44]: These seem to have the `[since_L4D]` and `[until_ASW]` still, intentional?

**Review by Gocnak — APPROVED** [review]
Great work! Went commit by commit and it didn't seem too bad. I noticed some L4D/ASW toggles still, but it otherwise looks good so it got thumbs. If you want to fix those you may, otherwise let me know if it's intentional and we can merge.

File: fgd/point/comp/comp_prop_cable.fgd (hunk 1 above)
Comment (braem) [reply fgd/point/comp/comp_prop_cable.fgd]: Woops! Not intentional no. Surprised it even compiled with that. I fixup'd those 2 commits (`comp_prop_cable` and `comp_prop_rope` had this).

Every field stayed except `screenspacefade`, as what we did with `prop_static` when removing ASW from the repo. It is marked as obsolete & shoots out a warning on Chaos.

**Review by Gocnak — APPROVED** [review]

**Pull request merged**